### ISSUE TITLE
Generated DeleteCollection methods make it impossible to use pagination correctly

### ISSUE THREAD

**Issue by DirectXMan12** [kubernetes/kubernetes#80877]
DeleteCollection supports the full set of list options -- under the hood, it just calls the same List & ListPredicate machinery in the Store implementation.  This means that it's perfectly viable to call DeleteCollection via REST calls directly with the pagination options (limit & continue), and, in fact, it behaves as expected when you do that (passing limit deletes a limitted set of results, returns those results, and gives back a continue token that can be passed to the next call to delete the next page of results).

However, our DeleteCollection generated Go methods don't actually return anything but an error, making it impossible to access the continue token, meaning that it's not possible, AFAICT, to use pagination correctly with the generated deletecollection methods.

Furthermore, since people look at the generated Go clients as a form of example/documention, we've effectively got misleading examples sitting around.

API calls tested against 1.13.7, relevant source code from k8s.io/apiserver & k8s.io/client-go examined @ HEAD-ish (bb707c8fe2e2b0b6ea8320e232f7725efda8de34)

/sig api-machinery
/kind bug

**Comment by DirectXMan12** [comment]
(the bugish-ness of this is debatable)

FYI @lavalamp @cheftako @roycaihw @rajathagasthya

**Comment by lavalamp** [comment]
@smarterclayton @wojtek-t @liggitt @deads2k 

Did we intend for DELETECOLLECTION to support pagination? Probably not? Maybe we should fix the server rather than letting the client make it easy to use the bug.

**Comment by liggitt** [comment]
Yeah, I wouldn't expect limit/continue to function externally for deletecollection (I could see storage using it internally to page through deletion)

**Comment by adohe** [comment]
cc @yue9944882

**Comment by wojtek-t** [comment]
> Yeah, I wouldn't expect limit/continue to function externally for deletecollection (I could see storage using it internally to page through deletion)

+1
